### PR TITLE
Changes default canvas_item_editor coarse zoom level from 1.5 to sqrt(2)

### LINKED ISSUE #25693: Restore 2d editor's zoom behavior from 3.0.6 back to a 2x / 0.5x multiplier
**Godot version:**

3.1 beta 3 (and many earlier versions of 3.1)

**Issue description:**
Sometime after 3.0.6 the zoom buttons in the 2D editor have changed from doubling/halving the zoom level to changing the zoom level at a rate of 1.5x:

44.4%, 66.7%, 100%, 150%, 225%, 337.5%, 506.3%

This has the effect of making it difficult to judge a game's graphics at an even ratio, which is particularly annoying when making a game without texture filtering enabled.

image

### PR DESCRIPTION
**Pull request by nobuyukinyuu**
Coarse zooming (using the +/- buttons in the canvas item editor) is changed to have even pixel ratios every other zoom level while providing intermediary zoom levels closer to 3.1 contributor preferences.  Doesn't change the 5% fine zoom when using the scroll wheel.  Simplest proposal to address https://github.com/godotengine/godot/issues/25693.  

*Bugsquad edit:* Fixes #25693.

### CODE REVIEW

**Comment by akien-mga** [comment]
Seems good to me. I suppose compilers are clever enough to optimize those `sqrt(2)` calls? Otherwise they could be replaced by `Math_SQRT2` from `core/math/math_defs.h`.

Does it properly snap to 25, 50, 100, 200, 400, etc. when reaching "round" values?

**Comment by nobuyukinyuu** [comment]
I only checked to 1 significant digit and it seemed to come out okay.  Since the function returned a double I figured it was probably fine enough for the purpose;  I tested it out with some pixel fonts and they looked fine at "even" zoom levels.  Provided it's a double, it might even still be fine to use a constant up to a zoom level most people could reasonably expect (maybe not 100x though!); I didn't know one was already defined, although changes to zoom is called infrequently enough that maybe it won't matter.

On "odd" values, the zoom level didn't come out even using sqrt(2).  The progression looked more like this:  100%, 141.4%, 200%, 282.8%, 400%, 565.7%, 800%, and so on.  I don't believe this is any worse than how it currently is, as it still seems to fit the original purpose, more-or-less, while restoring the even pixel ratio zoom levels.

**Comment by nobuyukinyuu** [comment]
I've updated this branch to use MATH_SQRT2 as it seemed to work just fine down to 12.5% and up to 1600%.  Hopefully that's a little bit better!  I don't know how to squash commits on git for the PR (I've seen this requested before while reading PRs) so if anything else needs to be done please let me know.

**Comment by akien-mga** [comment]
I'd ask you to squash the commits indeed :) See [PR workflow](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#modifying-a-pull-request).

TL;DR: `git rebase -i origin/master` and mark the second commit as `fixup` instead of `pick`. Then `git push --force origin 2d-editor-zoom-p1` to update the PR.

**Comment by groud** [comment]
I am not sure the implementation is ok, we need to be sure there are no way we have precision issues, else people might complain about their 2D pixel art not looking crisp enough.
Doing so likely requires implementing a zoom level using an integer, which would then be used to compute the final zoom value.

**Comment by groud** [comment]
Unless we implement that, I'd rather not merge it.

**Comment by nobuyukinyuu** [comment]
@groud The current implementation is already unacceptable for crisp pixel art.  I consider this a notable improvement, but was unwilling to implement it myself for lack of knowledge.  When I saw that none of the requests were going anywhere, I figured I'd give this PR a try.  Is there any reason to keep the current 1.5x zoom level over MATH_SQRT2?  It would certainly be nice to have 100% assurance of locking to integral zoom levels, but I don't expect anyone to take up that solution anytime soon...  (Don't ask me, I'm still new to this, lol)

**Comment by groud** [comment]
Ah indeed, you are right. I though the value was divided by two sorry.
Yes, then it is good to merge.

**Comment by nobuyukinyuu** [comment]
Thank you!

**Comment by nobuyukinyuu** [comment]
I've updated and hopefully have done the rebase correctly.

**Comment by akien-mga** [comment]
> I've updated and hopefully have done the rebase correctly.

Not exactly, as you now have 4 commits instead of 2. It seems like your properly squashed the two commits, but then you pulled your remote branch before pushing, so you applied the squashed commit on top of the previous 2 commits (+ a merge commit to glue this conflicting history).

**Comment by nobuyukinyuu** [comment]
I'm trying again, squashing those extra ones and resolving it, just having trouble authorizing the push from the console, since I usually only use SourceTree to do everything.  Once I figure out how to authorize from there I'll make the final force push.

**Comment by nobuyukinyuu** [comment]
The commits should be squashed now.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.2.
